### DISCUSSION TITLE
Migrate to TypeScript

### DISCUSSION THREAD

**Discussion by nyctfall** [jaredhanson/passport#962] [Polls]
Passport.js is a great library for all the underpinnings of authentication, especially for Express apps.

But it could be better!

One way it could be improved is with better type-checking and documentation.

Would those who are familiar with this repo and the Passport.js ecosystem please give their opinion on the matter?

**Comment by jaredhanson** [comment]
As mentioned in response to #961, I have no interest in converting this package to TypeScript.  As the maintainer, I wouldn't merge any PRs that make such changes.  I prefer to develop in JavaScript, and any language port wouldn't result in functional improvements.
